truth_set/java/structure/organisation/complex-code/ECommerce.java

```diff
@@ -0,0 +1,233 @@
+        return order != null && order.getCustomer() != null && 
+               order.getCustomer().getId() != null && 
+               order.getItems() != null && !order.getItems().isEmpty();
+    }
+
+    private void processOrderItems(OrderInput order, Map<String, Object> items) {
+        for (int i = 0; i < order.getItems().size(); i++) {
+            OrderItem item = order.getItems().get(i);
+            if (item.getProductId() == null || item.getQuantity() <= 0) {
+                continue;
+            }
+
+            double itemTotal = 0;
+            Double productPrice = getProductPrice(item.getProductId());
+            if (productPrice == null) {
+                items.put("item_" + i, "Price Not Available");
+                continue;
+            }
+
+            itemTotal += productPrice * item.getQuantity();
+
+            if (item.getOptions() != null) {
+                for (Option option : item.getOptions()) {
+                    if (option.getOptionId() == null || option.getValue() == null) {
+                        continue;
+                    }
+                    Double optionPrice = getOptionPrice(option.getOptionId(), option.getValue());
+                    if (optionPrice != null) {
+                        itemTotal += optionPrice;
+                    }
+                }
+            }
+
+            if (item.getDiscounts() != null) {
+                for (Discount discount : item.getDiscounts()) {
+                    if (discount.getDiscountId() == null || discount.getAmount() <= 0) {
+                        continue;
+                    }
+                    itemTotal -= discount.getAmount();
+                }
+            }
+
+            Map<String, Object> itemDetails = new HashMap<>();
+            itemDetails.put("productId", item.getProductId());
+            itemDetails.put("quantity", item.getQuantity());
+            itemDetails.put("total", Math.max(itemTotal, 0));
+            items.put("item_" + i, itemDetails);
+
+            if (order.getCoupons() != null) {
+                for (Coupon coupon : order.getCoupons()) {
+                    if (coupon.getCode() == null || !coupon.isValid() || 
+                        new Date(coupon.getExpiryDate()).before(new Date())) {
+                        continue;
+                    }
+                    ((Map<String, Object>)items.get("item_" + i)).put("coupon", coupon.getCode());
+                }
+            }
+        }
+
+        if (order.getShipping() != null && order.getShipping().getMethod() != null) {
+            if (order.getShipping().getCost() <= 0) {
+                finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + 
+                             ", Cost: Calculating");
+            } else {
+                finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + 
+                             ", Cost: $" + order.getShipping().getCost());
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting shipping message formatting logic</b></div>
<div id="fix">

Consider extracting the shipping message formatting logic into a separate method to improve code readability and reusability. The string concatenation for shipping details appears in multiple places.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -194,6 +194,14 @@
  private String formatShippingMessage(ShippingMethod method, double cost) {
      if (cost <= 0) {
          return "Method: " + method + ", Cost: Calculating";
      }
      return "Method: " + method + ", Cost: $" + cost;
  }

  if (order.getShipping().getCost() <= 0) {
 -    finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + ", Cost: Calculating");
 +    finalOrder.put("shipping", formatShippingMessage(order.getShipping().getMethod(), order.getShipping().getCost()));
  } else {
 -    finalOrder.put("shipping", "Method: " + order.getShipping().getMethod() + ", Cost: $" + order.getShipping().getCost());
 +    finalOrder.put("shipping", formatShippingMessage(order.getShipping().getMethod(), order.getShipping().getCost()));
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/8#issuecomment-2561208575>#e3503f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged